src/KimeraVioNode.cpp

```diff
@@ -0,0 +1,203 @@
+            std::bind(&VIO::RosDataProviderInterface::callbackMesherOutput,
+                      this,
+                      std::placeholders::_1));
+
+    bool use_lcd;
+    this->declare_parameter("use_lcd", false);
+    this->get_parameter("use_lcd", use_lcd);
+    if (use_lcd) {
+        vio_pipeline_.registerLcdOutputCallback(
+                std::bind(&VIO::RosDataProviderInterface::callbackLcdOutput,
+                          this,
+                          std::placeholders::_1));
+    }
+    handle_pipeline_ = std::async(std::launch::async,
+                                  &VIO::Pipeline::spin,
+                                  &vio_pipeline_);
+    handle_outputs_ = std::async(std::launch::async,
+                                  &KimeraVioNode::spin_outputs,
+                                  this);
+}
+
+bool KimeraVioNode::spin_outputs()
+{
+    while (rclcpp::ok()){
+        publishSyncedOutputs();
+        LOG_EVERY_N(INFO, 100) << "Done: publishSyncedOutputs()";
+        outputs_loop_rate_.sleep();
+
+        // Publish LCD output if any.
+        VIO::LcdOutput::Ptr lcd_output = nullptr;
+        if (lcd_output_queue_.pop(lcd_output)) {
+            publishLcdOutput(lcd_output);
+        }
+    }
+    return true;
+}
+
+KimeraVioNode::~KimeraVioNode()
+{
+    backend_output_queue_.shutdown();
+    frontend_output_queue_.shutdown();
+    mesher_output_queue_.shutdown();
+    lcd_output_queue_.shutdown();
+
+    vio_pipeline_.shutdown();
+    handle_pipeline_.get();
+    handle_outputs_.get();
+}
+
+void KimeraVioNode::stereo_cb(
+    const Image::SharedPtr left_msg,
+    const Image::SharedPtr right_msg)
+{
+  rclcpp::Time left_stamp(left_msg->header.stamp);
+  rclcpp::Time right_stamp(right_msg->header.stamp);
+  if(left_stamp.nanoseconds() > last_stereo_timestamp_.nanoseconds())
+  {
+  static const VIO::CameraParams& left_cam_info =
+      pipeline_params_.camera_params_.at(0);
+  static const VIO::CameraParams& right_cam_info =
+      pipeline_params_.camera_params_.at(1);
+
+  const VIO::Timestamp& timestamp_left = left_stamp.nanoseconds();
+  const VIO::Timestamp& timestamp_right = right_stamp.nanoseconds();
+
+  CHECK(left_frame_callback_)
+  << "Did you forget to register the left frame callback?";
+  CHECK(right_frame_callback_)
+  << "Did you forget to register the right frame callback?";
+//  TODO: Use RCLCPP_INFO inplace of CHECK?
+  // RCLCPP_INFO(this->get_logger(), "Did you forget to register the right frame callback?");
+
+  left_frame_callback_(VIO::make_unique<VIO::Frame>(
+      frame_count_, timestamp_left, left_cam_info, readRosImage(left_msg)));
+  right_frame_callback_(VIO::make_unique<VIO::Frame>(
+      frame_count_, timestamp_right, right_cam_info, readRosImage(right_msg)));
+  LOG_EVERY_N(INFO, 30) << "Done: KimeraVioNode::stereo_cb";
+  frame_count_++;
+}
+  last_stereo_timestamp_ = left_stamp;
+}
+
+void KimeraVioNode::imu_cb(const Imu::SharedPtr imu_msg)
```

Comment:
Would be nice if you could follow the repo's code style: `imuCb`

Reply:
Opps, will refactor; old happet from ROS1:
https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#function-and-method-naming